sdk/core/azure-core/inc/azure/core/credentials/credentials.hpp

```diff
@@ -77,7 +77,7 @@ namespace Azure { namespace Core { namespace Credentials {
   /**
    * @brief An exception that gets thrown when authentication error occurs.
    */
-  class AuthenticationException : public std::exception {
+  class AuthenticationException final : public std::exception {
```

Comment:
Maybe we should be making an exception for exceptions.

Reply:
We are starting to think of exception without seeing the need. I, on my side, do not think that customer code should be inheriting from our exceptions - customer code should rather be wrapping them.